### PR TITLE
docs: updated federation guide to include federation-specific graphql…

### LINKED ISSUE #3134: Add documentation for provided federated schema directives
Strawberry helpfully provides a bunch of federated schema directives we can use:
https://github.com/strawberry-graphql/strawberry/blob/628e4980b50e317cf99999aef052b8bb095370df/strawberry/federation/schema_directives.py

There's no reference to the provided directives in documentation, so I didn't even know they existed until I stumbled upon the code.

I can add documentation to at the very least list the defined directives and link out to apollo with details about what they are.

Where should this live?
In https://strawberry.rocks/docs/guides/federation?
In https://strawberry.rocks/docs/types/schema-directives?
In a new Federated Schema Directives page?

### PR DESCRIPTION
**Pull request by bradleyoesch**
Updated federation guide to include [Federation-specific GraphQL directives](https://www.apollographql.com/docs/federation/federated-types/federated-directives).

## Description

I didn't realize there were federated schema directives already implemented in strawberry! Updated documentation to make it easier to find.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

* Closes #3134

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by bradleyoesch** [comment]
Please leave any comments to clarify/change/update/etc!

File: docs/guides/federation.md [hunk 1]
```diff
@@ -323,6 +323,66 @@ We have provided a full example that you can run and tweak to play with
...
+- `@composeDirective`
+- `@external`
+- `@inaccessible`
+- `@override`
+- `@provides`
+- `@requires`
+- `@shareable`
+- `@tag`
+
+For example:
```
Comment (DoctorJohn) [inline docs/guides/federation.md:347]: nit:

```suggestion
For example, adding the following directives:
```

File: docs/guides/federation.md [hunk 2]
```diff
@@ -323,6 +323,66 @@ We have provided a full example that you can run and tweak to play with
...
+    id: strawberry.ID
+
+
+@strawberry.type(directives=[Shareable()])
+class CommonType:
+    foo: str
+    woops: bool = strawberry.field(directives=[Inaccessible()])
+```
+
+Will result in:
```
Comment (DoctorJohn) [inline docs/guides/federation.md:365]: nit:

```suggestion
Will result in the following GraphQL schema:
```

**Review by DoctorJohn — CHANGES_REQUESTED** [review]
Well written as well, thanks so much @bradleyoesch! I left some small comments which could give this PR it's final touch.

**Comment by bradleyoesch** [comment]
> Well written as well, thanks so much @bradleyoesch! I left some small comments which could give this PR it's final touch.

@DoctorJohn Thanks! I committed the suggestions

**Review by DoctorJohn — APPROVED** [review]
Awesome, thanks again for the PR! LGTM

**Review by patrick91 — APPROVED** [review]

**Pull request merged**